src/Estimators/include/BipedalLocomotion/FloatingBaseEstimators/ModelComputationsHelper.h

```diff
@@ -42,6 +42,46 @@ struct KinDynComputationsDescriptor
      * @return True if valid, false otherwise.
      */
     bool isValid() const;
+
+    /**
+     * Get the number of internal degrees of freedom of the robot model.
+     * It returns the internal degrees of freedom, because it does not include the eventual 6 degrees of freedom usually associated with the floating base.
+     * @return the number of internal degrees of freedom of the model.
+     */
+    int getNrOfDegreesOfFreedom() const;
+
```

Comment:
I don't know if this is the right way to proceed. If you need to access to the following methods in python you may:
1. Extend the pybind11 bindings in idyntree https://github.com/robotology/idyntree/tree/master/bindings/pybind11
2. Extend the pybind11 bindings of kindyn computations in blf 
    
    https://github.com/dic-iit/bipedal-locomotion-framework/blob/86aba53e9ad9a29084c4e58962b4f75daeba8b0a/bindings/python/FloatingBaseEstimators/src/FloatingBaseEstimators.cpp#L28-L29
   ```cpp
     py::class_<KinDynComputations, std::shared_ptr<KinDynComputations>>(module, 
                                                                     "KinDynComputations")
         .def("get_joint_pos", [](const KinDynComputations& impl) {
               Eigen::VectorXd s(kindyn.getNrOfDegreesOfFreedom());
               if (!impl.getJointPos(s))
               {
                    throw py::value_error("Failed to get the joint position.");
                }
               return s;
             })
         .def("set_joint_pos", [](const KinDynComputations& impl, Eigen::Ref<const Eigen::VectorXd> s) {
               return impl.setJointPos(iDynTree::make_span(s.data(), s.size()); 
             },
               py::arg("s"));
   ```


Probably @diegoferigo can give us some more hints

Reply:
All these methods are already exposed in the SWIG bindings of `KinDynComputations` (see their usage in the [high-level Python class](https://github.com/robotology/gym-ignition/blob/f1109734a78c8b3d695378c2070ed529ddb5ae1b/python/gym_ignition/rbd/idyntree/kindyncomputations.py) in gym-ignition). I'm still reviewing this PR and I might miss the rationale behind these new methods.

Reply:
Now that I finished the review, I think that the following option is better:

> 2. Extend the pybind11 bindings of kindyn computations in blf

As I commented in https://github.com/robotology/idyntree/pull/866, the pybind11 bindings are not the official Python bindings of iDynTree. Pretty much everywhere we use the SWIG-generated bindings, they are also what's included in the PyPI packages and, as soon as https://github.com/robotology/robotology-superbuild/issues/641 will be addressed, in conda-forge.

Reply:
I guess we should have a quick f2f or t2t alignment on this. I wonder if this is also related to https://github.com/dic-iit/bipedal-locomotion-framework/issues/305 .

Reply:
I'm available for any kind of follow up.

Reply:
Thanks for the suggestion, I followed 2 in https://github.com/dic-iit/bipedal-locomotion-framework/pull/303/commits/3d3653036c2a534ff2126f9bf924755c991ccb13. I am available as well for any further discussion.

Reply:
I will mark this conversation as solved for now, we can refer to it in case of any follow up